### PR TITLE
Widget: Widget.clear_widgets empty widget list does not remove all children

### PR DESCRIPTION
**Pull request by pydsigner**
Maintainer merge checklist
* [x] Title is descriptive/clear for inclusion in release notes.
* [x] Applied the `Component: core-widget` label.
* [x] Applied the `api-break` label?
* [ ] Applied the `release-highlight` label to be highlighted in release notes.
* [x] Added to the milestone version it was merged into.
* [x] **Unittests** are included in PR.
* [x] Properly documented, including `versionadded`, `versionchanged` as needed.

### CODE REVIEW

**Comment by pydsigner** [comment]
As noted in Discord discussion, this change *does* alter the API behavior, though not in a way that should break existing users.

**Comment by matham** [comment]
Good catch about the possibility of someone passing in `children`.

Could you add unittests? You don't have to, but it would be nice if you added tests.

**Comment by pydsigner** [comment]
> Good catch about the possibility of someone passing in `children`.

The case was being handled before, but not as well; if a list other than `children` was passed in, it would still be duplicated for no gain, and sequences other than lists (like iterators and sets) were being needlessly excluded.

> Could you add unittests? You don't have to, but it would be nice if you added tests.

Done.

File: kivy/tests/test_widget.py [hunk 1]
```diff
@@ -30,6 +30,23 @@ def test_invalid_add_widget(self):
...
+        root.add_widget(c2, index=1)
+        root.add_widget(c3, index=2)
+        self.assertEqual(root.children, [c1, c2, c3])
+        root.clear_widgets([c2])
+        self.assertEqual(root.children, [c1, c3])
+        root.clear_widgets([])
+        self.assertEqual(root.children, [c1, c3])
+        root.clear_widgets()
+        self.assertEqual(root.children, [])
+
```
Comment (matham) [inline kivy/tests/test_widget.py:49]:
```suggestion

    def test_clear_widgets_children(self):
        root = self.root
        for _ in range(10):
            root.add_widget(self.cls())
        self.assertEqual(len(root.children), 10)

        root.clear_widgets(root.children)
        self.assertEqual(root.children, [])
```

To make sure passing in `children` works.

**Review by matham — CHANGES_REQUESTED** [review]

**Review by matham — APPROVED** [review]

File: kivy/uix/widget.py [hunk 2]
```diff
@@ -684,11 +684,11 @@ def clear_widgets(self, children=None):
             The `children` argument can be used to specify the children you
             want to remove.
         '''
```
Comment (matham) [inline kivy/uix/widget.py:686]:
```suggestion
        .. versionchanged:: 2.1.0

            Specifying an empty ``children`` list leaves the widgets unchanged.
            Previously it was treated like ``None`` and all children were
            removed.
        '''
```

**Review by matham — CHANGES_REQUESTED** [review]
Forgot to document changes.

**Review by matham — APPROVED** [review]

**Pull request merged**